internal/grpc/grpc.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package grpc
+
+import (
+	"crypto/tls"
+	"errors"
+
+	grpcRetry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc/credentials/local"
+
+	"github.com/nginx/agent/v3/internal/config"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+)
+
+// these will come from the agent config
+var serviceConfig = `{
+	"loadBalancingPolicy": "round_robin",
+	"healthCheckConfig": {
+		"serviceName": "nginx-agent"
+	}
+}`
+var defaultCredentials = local.NewCredentials()
+
+func GetDialOptions(agentConfig *config.Config) []grpc.DialOption {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
```

Comment:
Are we sure that we wanna use the `WithBlock()` option?
```go
// WithBlock returns a DialOption which makes callers of Dial block until the
// underlying connection is up. Without this, Dial returns immediately and
// connecting the server happens in background.
//
// Use of this feature is not recommended.  For more information, please see:
// https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md
func WithBlock() DialOption {
	return newFuncDialOption(func(o *dialOptions) {
		o.block = true
	})
}
```

Reply:
removing based on recommendation. See next commit